### ISSUE TITLE
ncu does not propose updates if installed modules are up-to-date, even if package.json is behind.

### ISSUE THREAD

**Issue by mgohin** [tjunnone/npm-check-updates#263]
In my package.json

```
"angular": "^1.5.5"
```

Angular 1.5.8 is available but when running `ncu` (2.8.0) I'm getting 

```
All dependencies match the latest package versions :)
```

I'm missing something ?

**Comment by mgohin** [comment]
Allright, behavior is really misleading. It looks like ncu check in node_modules and do not purpose package.json updates if dependencies in node_modules are up to date.

I thought it would tell me that angular can be updated in package.json.

I deleted node_modules folder and then `ncu` told me

     angular             ^1.5.5  →   ^1.5.8

Keep it open to ask an option to get those informations even if deps in node_modules are UTD

**Comment by raineorshine** [comment]
I agree, this is misleading and has been a source of confusion for many users. I don't have immediate availability to work on this, but I'll put this as a priority when I do get time, or I will happily support someone who has a proposal of a better user experience and wants to submit a pull request.

**Comment by brettz9** [comment]
A workaround is to remove the semver ranges and hard-code the exact version.

**Comment by speige** [comment]
This was fixed by raineorshine on May 27 in this commit https://github.com/tjunnone/npm-check-updates/commit/71386c98ba581abf1c6ff88c1f6a86bfa06788ad (Thanks!!!). Unfortunately, it's slated for v3. I'm guessing because it could be considered a breaking change. However, I'd consider it a critical bug fix.

Since v3 won't get to npm for a long time, I've made a fork off master with this single fix merged in. 

You can reference it in your package.json like this:
"npm-check-updates": "github:speige/npm-check-updates"

I don't plan to maintain my fork, but npm-check-updates is unusable without this fix. I'll go back to the original repo/npm as soon as this is resolved.

Thanks raine & tomas! It's an awesome package!

**Comment by speige** [comment]
duplicate of #261 #389

**Comment by apaatsio** [comment]
Can this issue be closed if it's fixed in 71386c9? Same goes for the issues #261 and #389.

**Comment by speige** [comment]
Although this issue is "closed", it's not been published to npm (& appears it won't be until v3, which is a long ways out).

For those who don't want to wait, I've chosen to switch to npm-check:
```
	"scripts": {
		"update-npm-versions": "npm-check --skip-unused --update-all --save-exact --ignore",
		"update-npm-versions-interactive": "npm-check"
	},
	"devDependencies": {
		"npm-check": "5.8.0",
	}
```
